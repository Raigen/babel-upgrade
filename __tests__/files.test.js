const JSON5 = require('json5');
const fs = require('fs');
const pify = require('pify');
const crossSpawn = require('cross-spawn');
const rimraf = require('rimraf');

describe('file locator', function () {
  beforeAll(async () => {
    await pify(rimraf)('./filetests');
  });

  describe('babel-preset-env', function () {
    const dir = './filetests/babel-preset-env';

    beforeAll(async () => {
      const gitArgs = ['clone', 'https://github.com/babel/babel-preset-env', '--branch', '1.x', '--single-branch', dir];

      // pify only works with callback style
      // crossSpawn which uses child_process.spawn has no callback style
      await new Promise((resolve, reject) => {
        const child = crossSpawn('git', gitArgs, { stdio: 'inherit' });
        child.on('error', (err) => reject(err));
        child.on('close', () => resolve());
      });

      await new Promise((resolve, reject) => {
        const child = crossSpawn('node', ['../../bin/babel-upgrade'], { stdio: 'inherit', cwd: dir });
        child.on('error', (err) => reject(err, err.stack));
        child.on('close', () => resolve());
      });
    }, 10000);

    it('package.json', async function () {
      const rawPackage = (await pify(fs.readFile)(`${dir}/package.json`)).toString('utf8');
      const packageJson = JSON5.parse(rawPackage);

      expect(packageJson).toMatchSnapshot();
    });
  });

  describe('screeps-flowtype-jest-skeleton', function () {
    const dir = './filetests/screeps-flowtype-jest-skeleton';

    beforeAll(async () => {
      const gitArgs = ['clone', 'https://github.com/FossiFoo/screeps-flowtype-jest-skeleton', '--single-branch', dir];

      // pify only works with callback style
      // crossSpawn which uses child_process.spawn has no callback style
      await new Promise((resolve, reject) => {
        const child = crossSpawn('git', gitArgs, { stdio: 'inherit' });
        child.on('error', (err) => reject(err));
        child.on('close', () => resolve());
      });

      await new Promise((resolve, reject) => {
        const child = crossSpawn('node', ['../../bin/babel-upgrade'], { stdio: 'inherit', cwd: dir });
        child.on('error', (err) => reject(err, err.stack));
        child.on('close', () => resolve());
      });
    }, 10000);

    it('package.json', async function () {
      const rawPackage = (await pify(fs.readFile)(`${dir}/package.json`)).toString('utf8');
      const packageJson = JSON5.parse(rawPackage);

      expect(packageJson).toMatchSnapshot();
    });

    it('.babelrc', async function () {
      const rawBabelRC = (await pify(fs.readFile)(`${dir}/.babelrc`)).toString('utf8');
      const babelRC = JSON5.parse(rawBabelRC);

      expect(babelRC).toMatchSnapshot();
    });
  });

});
