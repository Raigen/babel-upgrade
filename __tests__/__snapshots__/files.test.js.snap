// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file locator babel-preset-env package.json 1`] = `
Object {
  "author": "Henry Zhu <hi@henryzoo.com>",
  "babel": Object {
    "env": Object {
      "test": Object {
        "plugins": Array [
          "istanbul",
        ],
      },
    },
    "presets": Array [
      Array [
        "@babel/preset-env",
        Object {
          "loose": true,
        },
      ],
    ],
  },
  "dependencies": Object {
    "@babel/core": "7.0.0-beta.40",
    "@babel/plugin-transform-arrow-functions": "7.0.0-beta.40",
    "@babel/plugin-transform-async-to-generator": "7.0.0-beta.40",
    "@babel/plugin-transform-block-scoped-functions": "7.0.0-beta.40",
    "@babel/plugin-transform-block-scoping": "7.0.0-beta.40",
    "@babel/plugin-transform-classes": "7.0.0-beta.40",
    "@babel/plugin-transform-computed-properties": "7.0.0-beta.40",
    "@babel/plugin-transform-destructuring": "7.0.0-beta.40",
    "@babel/plugin-transform-duplicate-keys": "7.0.0-beta.40",
    "@babel/plugin-transform-exponentiation-operator": "7.0.0-beta.40",
    "@babel/plugin-transform-for-of": "7.0.0-beta.40",
    "@babel/plugin-transform-function-name": "7.0.0-beta.40",
    "@babel/plugin-transform-literals": "7.0.0-beta.40",
    "@babel/plugin-transform-modules-amd": "7.0.0-beta.40",
    "@babel/plugin-transform-modules-commonjs": "7.0.0-beta.40",
    "@babel/plugin-transform-modules-systemjs": "7.0.0-beta.40",
    "@babel/plugin-transform-modules-umd": "7.0.0-beta.40",
    "@babel/plugin-transform-object-super": "7.0.0-beta.40",
    "@babel/plugin-transform-parameters": "7.0.0-beta.40",
    "@babel/plugin-transform-regenerator": "7.0.0-beta.40",
    "@babel/plugin-transform-shorthand-properties": "7.0.0-beta.40",
    "@babel/plugin-transform-spread": "7.0.0-beta.40",
    "@babel/plugin-transform-sticky-regex": "7.0.0-beta.40",
    "@babel/plugin-transform-template-literals": "7.0.0-beta.40",
    "@babel/plugin-transform-unicode-regex": "7.0.0-beta.40",
    "babel-plugin-syntax-trailing-function-commas": "^6.22.0",
    "babel-plugin-transform-es2015-typeof-symbol": "^6.23.0",
    "browserslist": "^2.1.2",
    "invariant": "^2.2.2",
    "semver": "^5.3.0",
  },
  "description": "A Babel preset for each environment.",
  "devDependencies": Object {
    "@babel/cli": "7.0.0-beta.40",
    "@babel/core": "7.0.0-beta.40",
    "@babel/helper-fixtures": "7.0.0-beta.40",
    "@babel/helper-plugin-test-runner": "7.0.0-beta.40",
    "@babel/preset-env": "7.0.0-beta.40",
    "@babel/register": "7.0.0-beta.40",
    "babel-eslint": "^7.1.1",
    "babel-plugin-istanbul": "^3.1.2",
    "chai": "^3.5.0",
    "codecov": "^1.0.1",
    "compat-table": "kangax/compat-table#957f1ff15972e8fb2892a172f985e9af27bf1c75",
    "electron-to-chromium": "^1.3.11",
    "eslint": "^3.17.1",
    "eslint-config-babel": "^6.0.0",
    "eslint-plugin-flowtype": "^2.29.1",
    "fs-extra": "~2.0.0",
    "lodash": "^4.17.4",
    "mocha": "^3.2.0",
    "nyc": "^10.1.2",
    "rimraf": "^2.6.1",
  },
  "homepage": "https://babeljs.io/",
  "license": "MIT",
  "main": "lib/index.js",
  "name": "babel-preset-env",
  "nyc": Object {
    "all": true,
    "include": Array [
      "src/*.js",
    ],
    "instrument": false,
    "sourceMap": false,
  },
  "repository": "https://github.com/babel/babel-preset-env",
  "scripts": Object {
    "build": "rimraf lib && babel src -d lib",
    "build-data": "node ./scripts/build-data.js",
    "changelog": "git log \`git describe --tags --abbrev=0\`..HEAD --pretty=format:' * %s (%an)' | grep -v 'Merge pull request'",
    "coverage": "BABEL_ENV=test nyc npm run test",
    "coverage-ci": "nyc report --reporter=json && codecov -f coverage/coverage-final.json",
    "dev": "babel -w src -d lib",
    "fix": "eslint . --fix",
    "lint": "eslint .",
    "prepublish": "npm run build",
    "test": "npm run build && npm run test-only",
    "test-ci": "nyc npm run test",
    "test-only": "mocha ./test --compilers js:@babel/register -t 10000",
  },
  "version": "1.6.1",
}
`;

exports[`file locator screeps-flowtype-jest-skeleton .babelrc 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-transform-flow-strip-types",
  ],
  "presets": Array [
    "@babel/preset-env",
    "@babel/preset-flow",
  ],
}
`;

exports[`file locator screeps-flowtype-jest-skeleton package.json 1`] = `
Object {
  "author": "Fossi",
  "dependencies": Object {
    "@babel/preset-flow": "7.0.0-beta.40",
  },
  "description": "",
  "devDependencies": Object {
    "@babel/cli": "7.0.0-beta.40",
    "@babel/core": "7.0.0-beta.40",
    "@babel/preset-env": "7.0.0-beta.40",
    "@babel/preset-flow": "7.0.0-beta.40",
    "babel-core": "^7.0.0-bridge.0",
    "babel-eslint": "^7.1.1",
    "babel-jest": "^18.0.0",
    "eslint": "^3.12.2",
    "eslint-config-screeps-neat-o": "^1.0.0",
    "eslint-plugin-flowtype": "^2.29.1",
    "flow-bin": "^0.37.0",
    "flow-cli": "0.0.0-pre",
    "install": "^0.8.2",
    "jest": "^18.0.0",
    "screeps-typescript-declarations": "^4.2.1",
    "typedef-converter": "^0.1.4",
  },
  "jest": Object {
    "automock": true,
    "cacheDirectory": "/tmp/<path>",
    "collectCoverage": true,
    "collectCoverageFrom": Array [
      "src/**/*.js",
    ],
    "notify": true,
    "setupFiles": Array [
      "./etc/jestsetup.js",
    ],
    "testPathDirs": Array [
      "test",
      "src",
    ],
    "verbose": true,
  },
  "main": "main.js",
  "name": "screeps",
  "scripts": Object {
    "test": "jest",
  },
  "version": "1.0.0",
}
`;
